tests/MongoDB.Driver.Tests/Specifications/Runner/MongoClientJsonDrivenSessionsTestRunner.cs

```diff
@@ -0,0 +1,162 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using MongoDB.Bson;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.TestHelpers.JsonDrivenTests;
+
+namespace MongoDB.Driver.Tests.Specifications.Runner
+{
+    public abstract class MongoClientJsonDrivenSessionsTestRunner : MongoClientJsonDrivenTestRunnerBase
+    {
+        private const string SessionIdKeySuffix = "__ClientSessionId";
+
+        protected override string[] ExpectedTestColumns => new[] { "description", "clientOptions", "failPoint", "sessionOptions", "operations", "expectations", "outcome", "async" };
+
+        // protected methods`
+        protected override void TestInitialize()
+        {
+            base.TestInitialize();
+            KillAllSessions();
+        }
+
+        protected override void AssertEvents(EventCapturer actualEvents, BsonDocument test)
+        {
+            base.AssertEvents(
+                actualEvents,
+                test,
+                document =>
+                {
+                    RecursiveFieldSetter.SetAll(
+                        document, 
+                        "lsid",
+                        value =>
+                        {
+                            if (ObjectMap.TryGetValue(value.AsString + SessionIdKeySuffix, out var sessionId))
+                            {
+                                return BsonValue.Create(sessionId);
+                            }
+                            else
+                            {
+                                return value;
+                            }
+                        });
+                });
+        }
+
+        protected override void ExecuteOperations(IMongoClient client, Dictionary<string, object> objectMap, BsonDocument test)
+        {
+            var newItems = new Dictionary<string, object>();
+            foreach (var mapItem in objectMap)
+            {
+                // Save session ids to have their values when the session will be disposed.
+                if (mapItem.Value is IClientSessionHandle clientSessionHandle)
```

Comment:
Save sessionId, since session object in a dictionary will be disposed in Assert step. And will not provide `SessionId` info.